src/core/server/elasticsearch/types.ts

```diff
@@ -55,6 +56,11 @@ export interface ElasticsearchServicePreboot {
  * @public
  */
 export interface ElasticsearchServiceSetup {
+  /**
+   * TODO: doc
+   */
+  setUnauthorizedErrorHandler: (handler: UnauthorizedErrorHandler) => void;
```

Comment:
TL;DR: the API looks good to me, thanks! 

-------------

As I mentioned in Slack, we can safely ignore `authResponseHeaders` since we use it only during Kerberos "handshake" that we don't want to perform in this case.

I also confirmed with the ES security team that during refresh the user info (== `authState`) should not change for Token, SAML, OIDC and Kerberos authentication (it's cached in `.security-tokens` index) which is great. 

The only case when user info can potentially change during refresh is PKI authentication since instead of using refresh token (we don't have one for PKI) we exchange existing client certificate to a new token. We know that certificate is the same (otherwise we should have caught that during authentication stage), and the chance that the role mappings have changed during access token lifetime (max 60 minutes) is very low.

Even though it'd be nice to update `authState` during refresh as well (for the consistency at least), considering all the above it feels safe enough to ignore it _assuming_ it requires significant effort to support (is that correct assessment?).